src/client/input/header/gyrospace.h

```diff
@@ -0,0 +1,393 @@
+/*
+ * =======================================================================
+ *
+ * Gyro Space to Play - A plug-and-play Gyro Space Transformer code
+ * Version 0.8.5
+ *
+ * Provides functionality for transforming gyro inputs into Local Space,
+ * Player Space, and World Space, while handling sensitivity adjustments,
+ * gravity alignment and dynamic orientation.
+ *  
+ * Compatible with both C and C++ environments.
+ *
+ * Based on the work by Jibb Smart (JoyShockMapper, GamepadMotionHelpers,
+ * Fortnite v.19.30's Gyro Aim/Flick Stick implementation)
+ *
+ * =======================================================================
+ */
+
+ #ifndef GYROSPACE_HPP
+ #define GYROSPACE_HPP
+ 
+ #ifdef __cplusplus
+ #include <cmath>
+ #include <cstdio>
+ #include <cstdbool>
+ #include <cstdint>
+ extern "C" {
+ #else
+ #include "math.h"
+ #include "stdbool.h"
+ #include "stdint.h"
+ #endif
+ 
+// Ensure s32 and u32 are correctly defined
+#ifndef s32
+    #define s32 int32_t
+#endif
+
+#ifndef u32
+    #define u32 uint32_t
+#endif
+
+// Define boolean types
+#ifndef bool
+    #define bool s32
+#endif
+
+#ifndef ubool
+    #define ubool u32
+#endif
```

Comment:
why do you define all this, it' not used (except for bool)?
what's the point of ubool?!

also: in C23 bool is a proper type (like int) so `#ifdef bool` doesn't work. but stdbool.h guarantees that bool is available anyway.